src/test/java/webdrivertasks/page/CalculatorPage.java

```diff
@@ -0,0 +1,163 @@
+package webdrivertasks.page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import webdrivertasks.waitings.WebElementWaiting;
+
+public class CalculatorPage {
+    private WebDriver driver;
+    private static final String EXTERNAL_IFRAME_XPATH = "//devsite-iframe/iframe";
+    private static final String FINAL_COST_LOCATOR = "//b[contains(text(),'Total Estimated Cost')]";
+    private static final String EXPANDED_DROPDOWN_XPATH = "//div[contains(@class,'clickable')]//div[contains(text(),'%s')]";
+
+    @FindBy(xpath = "//md-input-container/input[contains(@ng-model,'quantity')]")
+    private WebElement numberOfInstancesInputArea;
+
+    @FindBy(xpath = "//md-select[contains(@ng-model,'os')]")
+    private WebElement operatingSystemDropDown;
+
+    @FindBy(xpath = "//*[@placeholder='VM Class']")
+    private WebElement machineClassDropDown;
+
+    @FindBy(xpath = "//md-list[contains(@ng-repeat,'filterComputeItems')]//div[contains(text(),'VM class')]")
+    private WebElement currentMachineClass;
+
+    @FindBy(xpath = "//*[@placeholder='Instance type']")
+    private WebElement machineTypeDropDown;
+
+    @FindBy(xpath = "//md-list[contains(@ng-repeat,'filterComputeItems')]//div[contains(text(),'Instance type')]")
+    private WebElement currentInstanceType;
+
+    @FindBy(xpath = "//md-select[contains(@ng-model,'computeServer.location')]")
+    private WebElement dataCenterLocationDropDown;
+
+    @FindBy(xpath = "//md-select[contains(@ng-model,'computeServer.cud')]")
+    private WebElement machineCommitedUsageDropDown;
+
+    @FindBy(xpath = "//button[contains(@ng-click,'ComputeEngineForm')]")
+    private WebElement machineAddEstimationButton;
+
+    @FindBy(xpath = "//button[contains(@ng-click,'addSoleTenant')]")
+    private WebElement nodeAddEstimationButton;
+
+    @FindBy(xpath = "//input[@name='nodesCount']")
+    private WebElement numberOfNodesInputArea;
+
+    @FindBy(xpath = "//md-checkbox[contains(@ng-model,'addGPUs')]")
+    private WebElement addGPUCheckbox;
+
+    @FindBy(xpath = "//md-select[@placeholder='Number of GPUs']")
+    private WebElement numberOfGPUDropDown;
+
+    @FindBy(xpath = "//md-select[@placeholder='Local SSD']")
+    private WebElement localSSDDropDown;
+
+    @FindBy(xpath = "//md-select[contains(@ng-model,'soleTenant.cud')]")
+    private WebElement nodeCommitedUsageDropdown;
+
+    public CalculatorPage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public CalculatorPage inputNumberOfInstances(String numberOfInstances) {
+        WebElementWaiting.waitForPresenceOfElement(driver, EXTERNAL_IFRAME_XPATH);
+        driver.switchTo().frame(0);
```

Comment:
I'd suggest creating functions for switching to iframes. It could improve readability